pandas/core/generic.py

```diff
@@ -4545,26 +4545,23 @@ def pipe(self, func, *args, **kwargs):
 
     Parameters
     ----------
-    func : function, string, dictionary, or list of string/functions
+    func : function, str, list or dict
         Function to use for aggregating the data. If a function, must either
-        work when passed a %(klass)s or when passed to %(klass)s.apply. For
-        a DataFrame, can pass a dict, if the keys are DataFrame column names.
+        work when passed a %(klass)s or when passed to %(klass)s.apply.
 
         Accepted combinations are:
 
-        - string function name.
-        - function.
-        - list of functions.
-        - dict of column names -> functions (or list of functions).
+        - function
+        - string function name
+        - list of functions and/or function names, e.g. ``[np.sum, 'mean']``
+        - dict of axis labels -> functions, function names or list of such
     %(axis)s
-    *args
-        Positional arguments to pass to `func`.
-    **kwargs
-        Keyword arguments to pass to `func`.
+    *args, **kwargs
+        Arguments to pass to `func`.
 
     Returns
     -------
-    aggregated : %(klass)s
+    %(klass)s
```

Comment:
If you don't mind adding a description to what is returned, that would be great.

Reply:
The return type  is actually a bit more complex than expected:

```python
>>> df = pd.DataFrame({'A': range(5), 'B': 5})
>>> df.agg(['sum', 'mean'])  -> a DataFrame
>>>df.agg('sum')  -> a Series
>>> df.A.agg(['sum', 'mean'])  -> a Series
>>>df.A.agg('sum')  -> a scalar
```

I suggest we should do a ``%(return_value)s`` in the Returns section and add as appropriate in Series.agg and DataFrame.agg appenders.

Reply:
I'd simply have `DataFrame, Series or scalar` as the type, and then in the return description explain a bit better what is being returned in each case. May be there are better options, but this is what I've been doing, and I personally prefer to not overcomplicate the docstrings with variables, unless it really adds significant value. But that's my opinion, it's you call.